colossalai/zero/zero_optimizer.py

```diff
@@ -61,11 +58,20 @@ def __init__(self,
         assert isinstance(module, ZeroDDP)
         self.module = module
         self.gemini_manager = module.gemini_manager
-        self.chunk_manager = self.gemini_manager.chunk_manager
+        self.chunk_manager: ChunkManager = self.gemini_manager.chunk_manager
         self.optim_state = OptimState.UNSCALED
-        self.fp16_param_to_fp32_param: Dict[torch.Tensor, torch.Tensor] = {}
+        self.param_to_range: Dict[Parameter, Tuple[int, int]] = dict()
+        self.param_to_chunk32: Dict[Parameter, Chunk] = dict()
+        self.chunk16_set: Set[Chunk] = set()
+
         for p, fp32_p in zip(module.parameters(), module.fp32_params):
-            self.fp16_param_to_fp32_param[p] = fp32_p
+            chunk_16 = self.chunk_manager.get_chunk(p)
+            chunk_32 = self.chunk_manager.get_chunk(fp32_p)
+            chunk_32.init_pair(chunk_16)
```

Comment:
Why not init pair in ZeroDDP?

Reply:
It's a problem for a legacy reason. I'll move it to ZeroDDP.